### PR TITLE
general/bool: Remove TRUE/FALSE and boolean-style uses with int types

### PR DESCRIPTION
**Pull request by jlucovsky**
Describe changes:
- Remove libnss from CI steps
- Remove TRUE/FALSE and replace with logic/code using bool native types

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15408

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -2095,7 +2095,6 @@ jobs:
               libjansson-dev \
               libjansson4 \
               liblua5.1-dev \
-              libnss3-dev \
```
Comment (catenacyber) [inline .github/workflows/builds.yml:2098]: I like that :-)

**Comment by catenacyber** [comment]
Is there a ticket for this ?

**Comment by jlucovsky** [comment]
Continued in #9671

**Pull request closed without merging**